recirq/seniority_zero/circuits_expt/gates_this_experiment.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File to import all gates from to make changing gate sets easier"""
+
+gate_type = 'cz'
```

Comment:
this is really dirty lol. 

minimum bar: please use all caps variable name for `GATE_TYPE`, which is a global variable. Note that if you want to override it you must do so before importing this file. Please rename this file to something like `gate_type_import_delegation` or something weird and scary to denote that this is weird and scary.

A better way to structure this would be to set up a dataclass and two globals for the two gatesets


```python
@dataclass
class SeniorityZeroGateSet:
  cmnot: Gate
  cnot: Gate
  gsgate: Gate
  rzz: Gate
  swap_diag: Gate

SQRT_ISWAP_GATESET = SeniorityZeroGateSet(
  cmnot = cmnot_from_sqrt_swap,
 ..
)

CZ_GATESET = SeniorityZeroGateSet(
  cmnot = cmnot_from_cz,
  ...
)
```

Then to use it

```python
from .gate_sets import SeniorityZeroGateset, SQRT_ISWAP_GATESET, CZ_GATESET

def get_circuit_or_whatever(n: int, gs: SeniorityZeroGateset):
  for _ in range(n):
    circuit.append(gs.cmnot(i, j))

def driver_code():
   gateset = CZ_GATESET
   circuit = get_circuit(5, CZ_GATESET)
   ...
```

Reply:
Thanks for the tip, this is a far better idea than what I was doing. I'll push the change now and propagate it through with the latter imports.